test/sage/executor_test.exs

```diff
@@ -82,6 +82,48 @@ defmodule Sage.ExecutorTest do
     end
   end
 
+  test "async steps use the supplied supervisor if there is one" do
+    test_process = self()
+    ref = make_ref()
+
+    supervisor = ExUnit.Callbacks.start_supervised!({Task.Supervisor, name: Sage.MyTestSupervisor})
+
+    step = fn _, _ ->
+      ancestors = Process.info(self()) |> Keyword.fetch!(:dictionary) |> Keyword.fetch!(:"$ancestors")
+      send(test_process, {ref, ancestors})
+      {:ok, :ok}
+    end
+
+    new()
+    |> run_async(:step1, step, :noop, supervisor: supervisor)
+    |> execute(1)
+
+    # The async task will be awaited as it's the last step in the pipeline meaning
+    # there's no chance that we reach here _before_ the send has happened (causing a flaky test)
+    assert_received({^ref, ancestors}, 1_000)
```

Comment:
I would keep the default timeout here, if it starts to fail and become flaky, that's either a reason to increase it globally on CI (because of slow VMs) or there is something worth investigating.

I would also remove the comment above, I think await behavior is already explained in docs.